sky/cli.py

```diff
@@ -2449,13 +2449,16 @@ def spot_launch(
             f'Launching managed spot job {name} from spot controller...',
             fg='yellow')
         click.echo('Launching spot controller...')
-        sky.launch(dag,
-                   stream_logs=True,
-                   cluster_name=controller_name,
-                   detach_run=detach_run,
-                   idle_minutes_to_autostop=spot_lib.
-                   SPOT_CONTROLLER_IDLE_MINUTES_TO_AUTOSTOP,
-                   is_spot_controller_task=True)
+        sky.launch(
+            dag,
+            stream_logs=True,
+            cluster_name=controller_name,
+            detach_run=detach_run,
+            idle_minutes_to_autostop=spot_lib.
+            SPOT_CONTROLLER_IDLE_MINUTES_TO_AUTOSTOP,
+            is_spot_controller_task=True,
+            retry_until_up=True,
```

Comment:
Curious: why did the spot controller fail to be provisioned in the first try?

Reply:
I see. I think in such cases, either fail fast/once (old behavior) or retrying potentially a long time or indefinitely (this behavior) is justifiable.  I think this change is okay and we can keep an eye on user feedback.